src/Cli/Microsoft.DotNet.Cli.Utils/Extensions/OptionExtensions.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using System.Reflection;
+
+namespace Microsoft.DotNet.Cli.Utils.Extensions;
+
+public static class OptionExtensions
+{
+    private static readonly PropertyInfo[] s_nonPublicProperties = typeof(Option).GetProperties(BindingFlags.Instance | BindingFlags.NonPublic);
+
+    public static Argument? GetArgument(this Option option) =>
+        s_nonPublicProperties.First(pi => pi.Name == "Argument").GetValue(option) as Argument;
```

Comment:
This is the crux of a lot of pain. S.CL has a private Argument that drives the Option type (this is a standard composition design pattern). However, they don't expose everything from Argument that I need when referencing an Option. So, the only way to get that data is to access this Argument property in the Option. It is just a failing of API surface mirroring for a composed type.

Reply:
looking at how this is used, you really just want the HasDefaultValue/Default value stuff - and Option exposes all of that itself. So if an overload of WriteDefaultValue took an Option, that should be enough to render all of this reflection moot?